### ISSUE TITLE
[rush] Add rush audit command

### ISSUE THREAD

**Issue by antoine-coulon** [microsoft/rushstack#3727]
## Summary

One global command that is missing for Rush imho is the `rush audit` that would do exactly the same thing as `pnpm audit` or `yarn audit` and would allow us to perform a _software composition analysis_.

Currently, if I want to audit a project/library, I need to directly use the underlying package manager as mentioned above, which limits features that can be used in the context of Rush and is in my opinion an abstraction we want to preserve (in the same way as for all other commands).

**Current behavior**

To perform an _audit_ right now, I need to go in `common/config/rush` where my `pnpm-lock.yaml` is located, then I can run `pnpm audit`. 
Nevertheless, I'm kind of limited about features even when using the underlying package manager, for instance I'm not able to run `pnpm audit --fix` as the manifest target is not being resolved by `pnpm` itself.

**Expected behavior (with the feature)**

`rush audit` could be just run anywhere in the monorepo looking for apps/libs vulnerabilities. 

We could also imagine specifying: 
- **filters**: `pnpm` allows it under the hood (I don't know about other types of workspaces), such as `rush audit --targets=mylib1,mylib2`
- **severity levels**: `rush audit --targets=mylib1 --severity=critical`

Basically it would just proxy anything (or a subset) that supported package managers can do in terms of vulnerabilities audit.

We can discuss about what should be implemented for the first milestone, personally I feel like just running a global audit on the whole monorepo could be a great feature in the first place.

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | **5.79.0** |
| `rushVersion` from rush.json? | **5.79.0** |
| `useWorkspaces` from rush.json? | **true** |
| Would you consider contributing a PR? | **Yes** |
| Node.js version (`node -v`)? | **17.x.x** |

**Comment by D4N14L** [comment]
We'd be open to a PR adding this functionality if you're willing to provide one? `audit` is also a supported command for Yarn and NPM as well, so it's also something we could support for non-PNPM environments.

In the short-term, you can also use `rush-pnpm` to access this PNPM functionality, though filters would need to be specified using PNPM filter syntax.

**Comment by antoine-coulon** [comment]
Hello @D4N14L,

Yeah sure, I would be happy to provide a PR. 

Just before to start anything I want to be sure about what we expect from a first iteration about this `audit` command. As you mentioned, `rush-pnpm` allows most features of `pnpm` to be adapted using the Rush configuration. Do we just want to provide a simple interface over `audit` command of pnpm in the first place but by using `rush` binary over the `rush-pnpm` one with a subset of args that we find interesting for most cases?

Moreover as we both mentioned, `audit` is also a command that both `npm` and `yarn` provide, should I also cover those cases in the first PR?

`rush audit` would consequently be the command that either abstract `yarn audit`, `pnpm audit`, `npm audit` providing a mapping from Rush args to the ones expected by the package managers. 

@D4N14L @octogonz @iclanton any starting points / recommandations is welcome, I'll start a PR soon

Thanks guys

**Comment by D4N14L** [comment]
> Do we just want to provide a simple interface over `audit` command of pnpm in the first place but by using `rush` binary over the `rush-pnpm` one with a subset of args that we find interesting for most cases?

Yes that seems reasonable. This is essentially what `rush install` is (plus some other stuff that we add to make installs smoother in Rush).

> Moreover as we both mentioned, `audit` is also a command that both `npm` and `yarn` provide, should I also cover those cases in the first PR?

That's not explicitly necessary, no, though it would be appreciated :). The audit command when paired with those package installers should at least exit with a user-friendly error. Others may have other opinions on this though, cc @iclanton / @octogonz

**Comment by antoine-coulon** [comment]
@D4N14L fair enough, I'll start the work soon and given the discussion will stay open feel free to ping me there if you think about anything else :)
